Stage_3

KinCadeFoster opened the pull request:
- Вынес логику в фунции getUserStatistics и getDictionary
- Создал консольное меню будущего тренажера в цикле. Цикл обрабатывает ввод цифр 2 и 0
- При вводе чего угодно кроме 1, 2 или 0 – выводится сообщение "Неверный ввод. Введите доступные значения: 1, 2 или 0."
- Реализовал пункт меню “Статистика”, с выводом в консоль

KinCadeFoster committed:
Stage_3
- Вынес логику в фунции getUserStatistics и getDictionary
- Создал консольное меню будущего тренажера в цикле. Цикл обрабатывает ввод цифр 2 и 0
- При вводе чего угодно кроме 1, 2 или 0 – выводится сообщение "Неверный ввод. Введите доступные значения: 1, 2 или 0."
- Реализовал пункт меню “Статистика”, с выводом в консоль

danelyan commented on src/main/kotlin/Main.kt:
```diff
@@ -6,21 +6,62 @@ data class Word(
+    val dictionary = getDictionary()
+
+    while (true) {
+        println("Выберите пункт меню:")
```
можно обойтись одним вызовом и [многострочной строкой](https://kotlinlang.org/docs/strings.html#multiline-strings)

danelyan commented on src/main/kotlin/Main.kt:
```diff
@@ -6,21 +6,62 @@ data class Word(
+}
 
-fun main() {
+fun getDictionary(): MutableList<Word> {
```
Тут нам не нужен MutableList, старайся использовать минимально требуемый класс/интерфейс. В нашем случае List

danelyan commented on src/main/kotlin/Main.kt:
```diff
@@ -6,21 +6,62 @@ data class Word(
     val correctCount: Int,
 )
 
+fun getUserStatistics(dictionary: MutableList<Word>) {
```
тут достаточно List

KinCadeFoster committed:
Stage_3
- MutableList заменен на List
- Вывод информации о меню заменен на многострочный вариант вывода

danelyan commented on src/main/kotlin/Main.kt:
```diff
@@ -6,21 +6,63 @@ data class Word(
 
+fun getUserStatistics(dictionary: List<Word>) {
+    val wordCount = dictionary.size
+    val learnedWords = dictionary.count { it.correctCount > 2 }
```
тут тонкий момент, зачем нам считать learnedWords если dictionary пуст?
На практике этот count займет миллисекунды, но лучше положить его внутрь if

KinCadeFoster committed:
Stage_3
- Вынес подчет выученых слов внутрь ветки, что бы не считалось постоянно
